src/service.rs

```diff
@@ -250,6 +250,7 @@ where
 
                     _ if (modified || session_config.always_save)
                         && !empty
+                        && !res.status().is_client_error()
```

Comment:
Can you go into more detail regarding the logic here? [Django only filters out server errors](https://github.com/django/django/blob/6dcab75d5d8c2ef18de15323930057e6fa9ec00f/django/contrib/sessions/middleware.py#L55-L57).

Reply:
I did it here in the meantime:  https://github.com/maxcountryman/tower-sessions/pull/220#issuecomment-2436492905. ;)

Reply:
I've updated PR description.